aoc/d3/d3_test.go

```diff
@@ -0,0 +1,118 @@
+package main
+
+import "testing"
+
+// ensure the strings given get split in two and returned as
+// the first and second items in a slice
+func TestSplitString(t *testing.T) {
+	tests := []struct {
+		input   string
+		output1 string
+		output2 string
+	}{
+		{
+			input:   "abcd",
+			output1: "ab",
+			output2: "cd",
+		},
+		{
+			input:   "vJrwpWtwJgWrhcsFMMfFFhFp",
+			output1: "vJrwpWtwJgWr",
+			output2: "hcsFMMfFFhFp",
+		},
+		{
+			input:   "jqHRNqRjqzjGDLGLrsFMfFZSrLrFZsSL",
+			output1: "jqHRNqRjqzjGDLGL",
+			output2: "rsFMfFZSrLrFZsSL",
+		},
+	}
+
+	for _, test := range tests {
+		result := SplitString(test.input)
+		if result[0] != test.output1 {
+			t.Errorf("Got %s, expected %s", result, test.output1)
+		}
+		if result[1] != test.output2 {
+			t.Errorf("Got %s, expected %s", result, test.output2)
+		}
+	}
+}
+
+// ensure the function returns the character that appears in both strings
+func TestFindCommonItem(t *testing.T) {
+	tests := []struct {
+		input1   string
+		input2   string
+		expected string
+	}{
```

Comment:
Completely a matter of taste, you can also write
```suggestion
	tests := []struct {
		input1, input2, expected string
	}{
```

Reply:
Didn't know that! Less repetition is good 👍  Changed in https://github.com/lucasrijllart/advent-of-code-2022/pull/1/commits/85b685bb038e687cf47423e10f0b78c4b5ba021a